src/runtime/slice.go

```diff
@@ -51,3 +51,29 @@ func sliceCopy(dst, src unsafe.Pointer, dstLen, srcLen uintptr, elemSize uintptr
 	memmove(dst, src, n*elemSize)
 	return int(n)
 }
+
+// sliceGrow returns a new slice with space for at least newCap elements
+func sliceGrow(oldBuf unsafe.Pointer, oldLen, oldCap, newCap, elemSize uintptr) (unsafe.Pointer, uintptr, uintptr) {
```

Comment:
The code in this function is almost identical to the code in `sliceAppend`. Was this intentional? (Merging the two might be nice to avoid duplication, but might also increase complexity of the code so not sure).

Reply:
I think I originally had a comment here about the duplication. Upstream has this logic merged. So yes we can do that, but I'd rather get this patch unblocked in first.